src/test/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusQueryWiremockExtension.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.metering.service.prometheus;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import org.candlepin.subscriptions.prometheus.model.QueryResult;
+import org.junit.jupiter.api.extension.*;
+
+/**
+ * Utility class to initialize and stub a mocked version of the prometheus server using WireMock.
+ *
+ * <p>The test classes need to configure the prometheus url to use `localhost:9000`. Example:
+ * {@code @SpringBootTest(properties =
+ * "rhsm-subscriptions.metering.prometheus.client.url=http://localhost:9000")}
+ */
+public class PrometheusQueryWiremockExtension
```

Comment:
We can improve this extension by adding many features like mocking more Prometheus scenarios or allow to inject the PrometheusQueyWiremock directly into a test class field. 
However, I wanted to keep things simpler for now. Specially, if you prefer these tests being purely an unit test instead of an integration test.